pkg/registry/apps/replicaset/strategy.go

```diff
@@ -138,6 +138,7 @@ func (rsStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) f
 	oldReplicaSet := old.(*apps.ReplicaSet)
 
 	opts := pod.GetValidationOptionsFromPodTemplate(&newReplicaSet.Spec.Template, &oldReplicaSet.Spec.Template)
+	opts.AllowInvalidLabelValueInSelector = true
```

Comment:
why is this hard-coded to true? shouldn't it just use the value computed inside GetValidationOptionsFromPodTemplate?